src/pkg/services/m365/api/lists.go

```diff
@@ -143,6 +177,91 @@ func (c Lists) getListContents(ctx context.Context, siteID, listID string) (
 	return cols, cTypes, lItems, nil
 }
 
+func (c Lists) PostList(
+	ctx context.Context,
+	siteID string,
+	listName string,
+	oldListByteArray []byte,
+) (models.Listable, error) {
+	newListName := listName
+
+	oldList, err := BytesToListable(oldListByteArray)
+	if err != nil {
+		return nil, clues.WrapWC(ctx, err, "creating old list")
+	}
+
+	if name, ok := ptr.ValOK(oldList.GetDisplayName()); ok {
+		nameParts := strings.Split(listName, "_")
+		if len(nameParts) > 0 {
+			nameParts[len(nameParts)-1] = name
+			newListName = strings.Join(nameParts, "_")
+		}
+	}
+
+	// this ensure all columns, contentTypes are set to the newList
+	newList := ToListable(oldList, newListName)
+
+	// Restore to List base to M365 back store
+	restoredList, err := c.Stable.Client().Sites().BySiteId(siteID).Lists().Post(ctx, newList, nil)
```

Comment:
nit: prefer multiline on func chains for code readability.
```suggestion
	restoredList, err := c.Stable.
	  Client().
	  Sites().
	  BySiteId(siteID).
	  Lists().
	  Post(ctx, newList, nil)
```